js/src/forum/components/FlagList.js

```diff
@@ -26,11 +27,7 @@ export default class FlagList extends Component {
 
                 return (
                   <li>
-                    <a href={app.route.post(post)} className="Notification Flag" config={function(element, isInitialized) {
-                      m.route.apply(this, arguments);
-
-                      if (!isInitialized) $(element).on('click', () => app.flags.index = post);
```

Comment:
The reason the click listener was added via jquery was probably to prevent a redraw? Not sure if that's still relevant? Code is 5 years old... If it was still relevant we could use the new `event.redraw = false` Mithril v2 mechanism

Reply:
Would that prevent a redraw though? I've seen a few cases where event handlers (not just for click) were attached in config, and I've always assumed that the goal was to reduce the number of attrs.

Reply:
I've personally only ever done that to prevent Mithril triggering unnecessary redraws for events that I know are handled via direct DOM manipulation or things where redraws would bring the browser down (like scroll events), so that's why I was assuming that.

Re-reading the code, I think it might actually have been because defining `onclick` with `config=m.route` was not possible in Mithril v0.2 :thinking:

Reply:
Yes, please, let's prevent a redraw here.

Reply:
Why do we need to prevent a redraw here, and how did the previous code do so? The event handler didn't add `m.redraw.strategy = 'none'`

Reply:
Only Mithril's event handlers automatically trigger redraws. The previous code attached the event listener directly to the rendered DOM node using jQuery.

Reply:
Ah, I see. That makes sense, thanks!

Reply:
Doesn't the use of `route=""` and as such `m.route.Link` causes a redraw on each click anyway? Adding `event.redraw = false` might not actually do anything, it it might break the routing :sweat_smile: haven't tested.

> how did the previous code do so

The previous event handler was added through jQuery. there's no redraw logic on such events unless you call `m.redraw()`. On the other hand all `on<something>={}` event handlers added through Mithril automatically trigger a redraw, unless `event.redraw = false` is set. In Mithril v0.2 it was `m.redraw.strategy = 'none'`.